Google.PowerShell.IntegrationTests/BigQuery/BigQuery.BqJob.Tests.ps1

```diff
@@ -213,6 +213,125 @@ Describe "BqJob-Copy" {
+        $job.Configuration.Extract.PrintHeader | Should Be $false
+    }
+
+    It "should Extract/Load a basic CSV" {
+        $job = $table | Start-BqJob -Extract CSV "$gcspath/basic.csv" -Synchronous
+        $job.Status.State | Should Be "DONE"
+        $job.Status.ErrorResult | Should Be $null
+
+        $file = Get-GcsObject "ps_test_$r" "basic.csv"
+        $file | Should Not Be $null
+
+        $alt_tab = $test_set | New-BqTable "basic_test_$r"
+        New-BqSchema -Name "Title" -Type "STRING" | New-BqSchema -Name "Author" -Type "STRING" |
+            New-BqSchema -Name "Year" -Type "INTEGER" | Set-BqSchema $alt_tab
+        $boj = $alt_tab | Start-BqJob -Load CSV "$gcspath/basic.csv" -SkipLeadingRows 1 -Synchronous
+        $boj.Status.State | Should Be "DONE"
+
+        $alt_tab = $alt_tab | Get-BqTable
+        $alt_tab.NumRows | Should Be 10
+    }
+
+    It "should Extract/Load a more complex CSV request" {
+        $job = $table | Start-BqJob -Extract CSV "$gcspath/complex.csv" -FieldDelimiter "|" -NoHeader -Synchronous
+        $job.Status.State | Should Be "DONE"
+        $job.Status.ErrorResult | Should Be $null
+
+        $file = Get-GcsObject "ps_test_$r" "basic.csv"
+        $file | Should Not Be $null
+
+        $alt_tab = $test_set | New-BqTable "complex_test_$r"
+        New-BqSchema -Name "Title" -Type "STRING" | New-BqSchema -Name "Author" -Type "STRING" |
+            New-BqSchema -Name "Year" -Type "INTEGER" | Set-BqSchema $alt_tab
+        $boj = $alt_tab | Start-BqJob -Load CSV "$gcspath/complex.csv" -FieldDelimiter "|" -Synchronous
+        $boj.Status.State | Should Be "DONE"
+
+        $alt_tab = $alt_tab | Get-BqTable
+        $alt_tab.NumRows | Should Be 10
+    }
+
+    It "should handle WriteMode correctly" {
+        $table | Start-BqJob -Extract CSV "$gcspath/write.csv" -Synchronous
+        $file = Get-GcsObject "ps_test_$r" "write.csv"
+        $alt_tab = $test_set | New-BqTable "writemode_test_$r"
+        New-BqSchema -Name "Title" -Type "STRING" | New-BqSchema -Name "Author" -Type "STRING" |
+            New-BqSchema -Name "Year" -Type "INTEGER" | Set-BqSchema $alt_tab
+
+        $alt_tab | Start-BqJob -Load CSV "$gcspath/write.csv" -SkipLeadingRows 1 `
+            -WriteMode WriteIfEmpty -Synchronous
+        $alt_tab = $alt_tab | Get-BqTable
+        $alt_tab.NumRows | Should Be 10
+
+        $alt_tab | Start-BqJob -Load CSV "$gcspath/write.csv" -SkipLeadingRows 1 `
+            -WriteMode WriteAppend -Synchronous
+        $alt_tab = $alt_tab | Get-BqTable
+        $alt_tab.NumRows | Should Be 20
+
+        $alt_tab | Start-BqJob -Load CSV "$gcspath/write.csv" -SkipLeadingRows 1 `
+            -WriteMode WriteTruncate -Synchronous
+        $alt_tab = $alt_tab | Get-BqTable
+        $alt_tab.NumRows | Should Be 10
+    }
+
+    It "Load should properly halt when -WhatIf is passed" {
+        $table | Start-BqJob -Load AVRO "$gcspath/whatif.json" -WhatIf | Should Be $null
```

Comment:
-WhatIf is a PS [builtin](https://msdn.microsoft.com/en-us/library/system.management.automation.cmdletcommonmetadataattribute.supportsshouldprocess(v=vs.85).aspx) that does effectively the same thing as the dry run flag of BqJobs - it outputs to the user some information about what would happen if this cmd would run, so they can judge whether it has the intended consequences.  

I added it to the cmdlet by specifying:
`[Cmdlet("Start", "BqJob", SupportsShouldProcess = true)]`